FindOnPage.md

```diff
@@ -0,0 +1,478 @@
+}
+
+//! [GetActiveMatchIndex]
+```
+
+
+
+## API Details
+```csharp
+
+/// Specifies the direction of find Parameters.
+[v1_enum]
+typedef enum COREWEBVIEW2_FIND_DIRECTION { 
+    /// Specifies a forward find in the document.
+    COREWEBVIEW2_FIND_DIRECTION_FORWARD, 
+    /// Specifies a backwards find in the document.
+    COREWEBVIEW2_FIND_DIRECTION_BACKWARD 
+} COREWEBVIEW2_FIND_DIRECTION; 
+
+// Interface defining the find configuration.
+[uuid(4A6ED732-DF08-4449-8949-3632A4DEBFCD), object, pointer_default(unique)] 
+interface ICoreWebView2FindConfiguration : IUnknown {
+
+    // Gets or sets the find term used for the find operation.
+    [propget] HRESULT FindTerm([out, retval] LPWSTR* value);
+    [propput] HRESULT FindTerm([in] LPCWSTR value);
+
+    // Gets or sets the direction of the find operation (forward or backward).
+    [propget] HRESULT FindDirection([out, retval] COREWEBVIEW2_FIND_DIRECTION* value); 
+    [propput] HRESULT FindDirection([in] COREWEBVIEW2_FIND_DIRECTION value);
+
+    // Gets or sets whether the find operation is case sensitive.
+    [propget] HRESULT IsCaseSensitive([out, retval] BOOL* value); 
+    [propput] HRESULT IsCaseSensitive([in] BOOL value);
+
+    // Gets or sets whether to only match whole words during the find operation.
+    [propget] HRESULT ShouldMatchWord([out, retval] BOOL* value); 
+    [propput] HRESULT ShouldMatchWord([in] BOOL value);
+
+    // Gets the state of whether all matches are highlighted.
+    // Returns TRUE if all matches are highlighted, FALSE otherwise.
+    [propget] HRESULT ShouldHighlightAllMatches([out, retval] BOOL* value); 
+    // Sets the state to either highlight all matches or not.
+    [propput] HRESULT ShouldHighlightAllMatches([in] BOOL value);
+ 
+    // Checks if a custom user interface is desired by end developer.
+    // If TRUE, the default Find bar is not displayed.
+    // Returns TRUE if using a custom UI, FALSE if using the default.
+    [propget] HRESULT SuppressDefaultDialog([out, retval] BOOL* value); 
+    // Sets whether to use a custom UI for the Find operation.
+    [propput] HRESULT SuppressDefaultDialog([in] BOOL value);
+
+    // Gets the index of the currently active match.
+    // If there's no active find session but an attempt is made to get the active match index:
+    // The function will return -1.
+    // Returns The active match index.
+    [propget] HRESULT ActiveMatchIndex([out, retval] LONG* value); 
+
+    // Gets the total count of matches found in the current document based on the last find criteria.
+    // Returns the total count of matches.
+    [propget] HRESULT MatchesCount([out, retval] LONG* value);
+
+}
+
+/// Handles the event that's fired when the match count changes.
+[uuid(623EFBFB-A19E-43C4-B309-D578511D24AB), object, pointer_default(unique)]
+interface ICoreWebView2FindMatchCountChangedEventHandler : IUnknown {
+    /// Parameter is the match count.
+    HRESULT Invoke(LONG matchesCount);
```

Comment:
Event handler interface Invoke methods have two parameters, the first is the source of the event (the ICoreWebView2Find object in this case) and the second is the event args interface. For [Property]Changed events like this we don't need an event args interface at all and just use IUnknown and nullptr for the event args because rather than duplicate the changed value in the event args the end dev can just as easily access the property on the ICoreWebView2Find interface.

```suggestion
    HRESULT Invoke(ICoreWebView2Find* source, IUnknown* eventArgs);
```

Unless it turns out that the properties actually need to be Get..Async async methods and not properties in which case we would want an ICoreWebView2FindMatchCountChangedEventArgs interface with a MatchesCount property on it.